src/main/java/com/internship/training/StreamsChallenge.java

```diff
@@ -79,31 +85,47 @@ private void getBest10SeriesByRating(){
         System.out.println("------------------------------------------------------------");
         System.out.println("These are the top 10 series:");
         //TODO print the name of the top 10 series - one by line
-
-        System.out.println("------------------------------------------------------------");
+       series.stream()
+               .sorted(Comparator.comparing(Series::getRating).reversed())
+               .limit(10)
+               .map(Series::getName)
+               .forEach(System.out::println);
+       System.out.println("------------------------------------------------------------");
     }
 
     private void getAllGenres(){
         System.out.println("------------------------------------------------------------");
         System.out.println("These are all the genres found:");
         //TODO print all the genres of the series sorted alphabetically, they can not be repeated - one by line
-
+        List<String> genres = series.stream()
+                .map(Series::getGenres)
+                .flatMap(Collection::stream)
+                .sorted()
+                .distinct()
+                .collect(Collectors.toList());
+        System.out.print(genres);
         System.out.println("------------------------------------------------------------");
     }
 
     private void getSeriesByStudioShaft(){
         System.out.println("------------------------------------------------------------");
         System.out.println("Series by Studio Shaft:");
         //TODO print the name of all Studio 'Shaft' series - one by line
-
+        series.stream()
+                .filter(s -> s.getStudios().contains("Shaft"))
+                .forEach(thisSeries -> System.out.print(""+"--"+thisSeries.getName()+"--"+" "));
```

Comment:
this will not print each series in its own line
you can do a map for the series name and then a forEach(System.out::println)